### ISSUE TITLE
NoMethodError: undefined method `[]' for nil:NilClass

### ISSUE THREAD

**Issue by qub3r** [ankane/searchkick#1007]
gems: 
searchkick v1.5.1
elasticsearch v1.0.17
mongoid v4.0.2
DBs:
elasticsearch v1.13.x
mongo v2.6.x

When I `bundle exec rake searchkick:reindex CLASS=Publisher` the Publisher class seems to rebuild the index ok.

However, when I attempt to `bundle exec rake searchkick:reindex:all` I always hit the error below on the Publisher class and I'm having trouble figuring out why.
```
rake aborted!
NoMethodError: undefined method `[]' for nil:NilClass
/opt/gems/gems/searchkick-1.5.1/lib/searchkick/index_options.rb:5:in `index_options'
/opt/gems/gems/searchkick-1.5.1/lib/searchkick/model.rb:81:in `searchkick_index_options'
/opt/gems/gems/searchkick-1.5.1/lib/searchkick/index.rb:201:in `reindex_scope'
/opt/gems/gems/searchkick-1.5.1/lib/searchkick/model.rb:58:in `searchkick_reindex'
/opt/gems/gems/searchkick-1.5.1/lib/searchkick/tasks.rb:26:in `block (4 levels) in <top (required)>'
/opt/gems/gems/searchkick-1.5.1/lib/searchkick/tasks.rb:24:in `each'
/opt/gems/gems/searchkick-1.5.1/lib/searchkick/tasks.rb:24:in `block (3 levels) in <top (required)>'
Tasks: TOP => searchkick:reindex:all
(See full trace by running task with --trace)
```

Any help would be appreciated.

**Comment by qub3r** [comment]
After some more troubleshooting, this seems to happen in `task.rb` after `Rails.application.eager_load!`. 

Looping through an arrary of models to reindex, without running eager_load!, works fine.
```
[Model, Model, Model].each do |model|
  model.reindex
end
```

Any ideas why eager_load would break one of our models?

**Comment by ankane** [comment]
Hey @qub3r, not really sure. From the error, it looks like options are set to `nil` somehow.

**Comment by qub3r** [comment]
We upgraded to ES5 and searchkick 2.3.2. Still getting the same issue.

I see what you mean by options are nil. This particular model is our Publisher model. When I Publisher.searchkick_index_options in the rails console, I get the same error.

I discovered the Publisher model isn't in the standard directory where you would find models (app/model). Instead it's located under app/modules. When I move the Publisher model back to the app/model directory, searchkick works fine again. At this point it definitely doesn't seem like a searchkick issue.
